shared/actions/signup/index.js

```diff
@@ -242,8 +242,12 @@ function signup (skipMail): TypedAsyncAction<Signup | ShowPaperKey> {
   })
 }
 
-export function resetSignup (): ResetSignup {
-  return {type: Constants.resetSignup, payload: {}}
+export function resetSignup (): TypedAsyncAction<ResetSignup | RouteAppend> {
+  return dispatch => new Promise((resolve, reject) => {
+    dispatch({type: Constants.resetSignup, payload: {}})
+    dispatch(nextPhase())
```

Comment:
i really dislike this model of having to call nextPhase, we're always affecting what the phase should be, this should be implied